spec/api/v2/commercial/organizations_spec.rb

```diff
@@ -0,0 +1,244 @@
+        post '/api/v2/commercial/organizations',
+             headers: auth_header
+        expect_status_to_eq 422
+      end
+    end
+
+    context 'when params is valid' do
+      let(:test_user) { User.find(1) }
+
+      it 'deny to create organization for normal user' do
+        post '/api/v2/commercial/organizations',
+             params: { name: 'Company Test', group: 'vip-1' },
+             headers: auth_header
+
+        expect(response.status).to eq(404)
+      end
+
+      it 'create organization for barong organization admin' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        post '/api/v2/commercial/organizations',
+             params: { name: 'Company Test', group: 'vip-1' },
+             headers: auth_header
+
+        expect(response.status).to eq(201)
+      end
+    end
+  end
+
+  describe 'GET /api/v2/commercial/organization' do
+    context 'when params is missing' do
+      it 'renders an error' do
+        get '/api/v2/commercial/organization', headers: auth_header
+
+        expect_status_to_eq 422
+      end
+    end
+
+    context 'find organization by oid' do
+      let(:do_request) { get '/api/v2/commercial/organization', headers: auth_header, params: { oid: 'OID001' } }
+      let(:test_user) { User.find(1) }
+
+      it 'deny request from normal user' do
+        do_request
+
+        expect(response.status).to eq(404)
+      end
+
+      it 'return organization details for barong organization admin' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        do_request
+
+        expect(response.status).to eq(200)
+      end
+    end
+  end
+
+  describe 'POST /api/v2/commercial/organization/update' do
+    context 'when params is missing' do
+      it 'renders an error' do
+        post '/api/v2/commercial/organization/update', headers: auth_header
+
+        expect_status_to_eq 422
+      end
+    end
+
+    context 'user is normal user' do
+      let(:do_request) do
+        post '/api/v2/commercial/organization/update',
+             headers: auth_header,
+             params: { oid: 'OID001' }
+      end
+      let(:test_user) { User.find(1) }
+
+      it 'deny request from normal user' do
+        do_request
+
+        expect(response.status).to eq(404)
+      end
+    end
+
+    context 'user is barong admin organization user' do
+      let(:params) { { oid: 'OID001' } }
+      let(:do_request) do
+        post '/api/v2/commercial/organization/update',
+             headers: auth_header,
+             params: params
+      end
+      let(:test_user) { User.find(1) }
+
+      it 'update organization field' do
+        # Assign user as barong organization admin
+        create(:membership, id: 1, user_id: 1, organization_id: 0)
+
+        params[:name] = 'Company Test'
+        do_request
+
+        expect(response.status).to eq(200)
+      end
+
```

Comment:
database state is not checked, cannot ensure request did anything

Reply:
testcase with several params changed is missing, and from the code it looks like A\pi can only update one key at a time

Reply:
Correct, This endpoint is allow only 1 param to update b/c the UI don't have save btn to make save all params at a time.

Reply:
I will do more check to ensure request was successfully